d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -201,16 +210,17 @@ public void getClient(Request request, RequestContext requestContext, Callback<T
         }
         else
         {
-          _log.debug("service hint found, using generic client for target: {}", targetService);
-
-          TransportClient transportClient = _state.getClient(serviceName, targetService.getScheme());
+          URI target = override == null ? targetService : URI.create(override + service.getPath());
+          _log.debug("Host override found, using generic client for target: {}", target);
```

Comment:
Eventhough TargetService hint is deprecated - should we include that int he debug (Host override/TargetService Hint found) until we completely remove the TargetService?

Reply:
Added as suggested.